src/Particular.ThroughputCollector.Persistence/IThroughputDataStore.cs

```diff
@@ -15,11 +15,14 @@ public interface IThroughputDataStore
 
     Task SaveEndpoint(Endpoint endpoint, CancellationToken cancellationToken = default);
 
-    Task RecordEndpointThroughput(EndpointIdentifier id, IEnumerable<EndpointDailyThroughput> throughput, CancellationToken cancellationToken = default);
+    Task<IDictionary<string, IEnumerable<ThroughputData>>> GetEndpointThroughputByQueueName(IEnumerable<string> queueNames, CancellationToken cancellationToken = default);
 
-    Task UpdateUserIndicatorOnEndpoints(List<Endpoint> endpointsWithUserIndicator);
+    Task RecordEndpointThroughput(string endpointName, ThroughputSource throughputSource, DateOnly date, long messageCount, CancellationToken cancellationToken = default) =>
+        RecordEndpointThroughput(endpointName, throughputSource, [new EndpointDailyThroughput(date, messageCount)], cancellationToken);
+
+    Task RecordEndpointThroughput(string endpointName, ThroughputSource throughputSource, IEnumerable<EndpointDailyThroughput> throughput, CancellationToken cancellationToken = default);
```

Comment:
Shouldn't the method take an `EndpointIdentifier`?

Reply:
@abparticular not sure what you mean here by "see above"?

Reply:
It seems the comment order has changed from the file view I working on.  I mean https://github.com/Particular/ServiceControl/pull/4052#discussion_r1550774091